src/Symfony/Component/Process/Process.php

```diff
@@ -1530,14 +1530,15 @@ private function requireProcessIsTerminated(string $functionName)
     /**
      * Escapes a string to be used as a shell argument.
      */
-    private function escapeArgument(string $argument): string
+    private function escapeArgument($argument): string
     {
+        $argument = (string) $argument;
+        if ('' === $argument) {
+            return '""';
+        }
         if ('\\' !== \DIRECTORY_SEPARATOR) {
             return "'".str_replace("'", "'\\''", $argument)."'";
         }
-        if ('' === $argument = (string) $argument) {
```

Comment:
I'm not sure this change is needed. Should be reverted to me. Only the type hint needs to be changed to me.

Reply:
Isn't it more optimal to first check for emptiness?

Reply:
I didn't remove the lines, just moved the up.

Reply:
It's unrelated and not needed for what you want to fix isn't it?

Reply:
agree that it's unrelated :)